vm/src/obj/objstr.rs

```diff
@@ -312,11 +310,11 @@ impl PyString {
 
     #[pymethod(name = "__hash__")]
     fn hash(&self) -> pyhash::PyHash {
-        match self.hash.get() {
+        match self.hash.load() {
```

Comment:
Here you should probably use `compare_and_swap`

Reply:
I think it's fine, since there's not really any stake to the value here; if 2 threads calculate and set it simultaneously than it'll be fine, and for `compare_and_swap` we have to provide a new value anyway, which defeats the purpose of a hash cache.